Replace disambiguation strings with translator comments

hebasto opened the pull request:
Since https://github.com/bitcoin/bitcoin/pull/21694 is merged, translator comments is the right way to pass context to translators.

This PR fixes changes were made:
- in #220 before https://github.com/bitcoin/bitcoin/pull/21694
- in https://github.com/bitcoin/bitcoin/pull/21694 on testing purpose
- in #125

Closes #288.

jarolrod commented on src/qt/addressbookpage.cpp:
```diff
@@ -281,7 +281,9 @@ void AddressBookPage::on_exportButton_clicked()
     QString filename = GUIUtil::getSaveFileName(this,
         tr("Export Address List"), QString(),
-        tr("Comma separated file", "Name of CSV file format") + QLatin1String(" (*.csv)"), nullptr);
+        /*: Name of CSV file format.
```
```suggestion
        /*: Expanded name of the CSV file format.
```

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/332#issuecomment-841732322).

jarolrod commented on src/qt/intro.cpp:
```diff
@@ -385,7 +385,9 @@ void Intro::UpdatePruneLabels(bool prune_checked)
     const uint64_t expected_backup_days = m_prune_target_gb * 1e9 / (uint64_t(expected_block_data_size) * 86400 / nPowTargetSpacing);
-    ui->lblPruneSuffix->setText(tr("(sufficient to restore backups %n day(s) old)", "block chain pruning", expected_backup_days));
+    ui->lblPruneSuffix->setText(
+        //: A note about a pruned node capability.
```
```suggestion
        //: Explanatory text on the capability of the current prune target.
```

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/332#issuecomment-841732322).

jarolrod commented on src/qt/psbtoperationsdialog.cpp:
```diff
@@ -141,7 +141,8 @@ void PSBTOperationsDialog::saveTransaction() {
     QString filename = GUIUtil::getSaveFileName(this,
         tr("Save Transaction Data"), filename_suggestion,
-        tr("Partially Signed Transaction (Binary)", "Name of binary PSBT file format") + QLatin1String(" (*.psbt)"), &selected_filter);
+        //: Name of binary PSBT file format. See: BIP 174.
```
```suggestion
        //: Expanded name of the binary PSBT file format. See: BIP 174.
```

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/332#issuecomment-841732322).

jarolrod commented on src/qt/qrimagewidget.cpp:
```diff
@@ -118,7 +118,9 @@ void QRImageWidget::saveImage()
     QString fn = GUIUtil::getSaveFileName(
         this, tr("Save QR Code"), QString(),
-        tr("PNG Image", "Name of PNG file format") + QLatin1String(" (*.png)"), nullptr);
+        /*: Name of PNG file format.
```
```suggestion
        /*: Expanded name of the PNG file format.
```

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/332#issuecomment-841732322).

jarolrod commented on src/qt/sendcoinsdialog.cpp:
```diff
@@ -432,7 +432,8 @@ void SendCoinsDialog::sendButtonClicked([[maybe_unused]] bool checked)
             QString filename = GUIUtil::getSaveFileName(this,
                 tr("Save Transaction Data"), fileNameSuggestion,
-                tr("Partially Signed Transaction (Binary)", "Name of binary PSBT file format") + QLatin1String(" (*.psbt)"), &selectedFilter);
+                //: Name of binary PSBT file format. See: BIP 174.
```
```suggestion
                //: Expanded name of the binary PSBT file format. See: BIP 174.
```

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/332#issuecomment-841732322).

jarolrod commented on src/qt/transactionview.cpp:
```diff
@@ -336,7 +336,9 @@ void TransactionView::exportClicked()
     QString filename = GUIUtil::getSaveFileName(this,
         tr("Export Transaction History"), QString(),
-        tr("Comma separated file", "Name of CSV file format") + QLatin1String(" (*.csv)"), nullptr);
+        /*: Name of CSV file format.
```
```suggestion
        /*: Expanded name of the CSV file format.
```

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/332#issuecomment-841732322).

jarolrod commented on src/qt/addressbookpage.cpp:
```diff
@@ -295,8 +297,9 @@ void AddressBookPage::on_exportButton_clicked()
         QMessageBox::critical(this, tr("Exporting Failed"),
-            //: %1 is a name of the file (e.g., "addrbook.csv") that the bitcoin addresses were exported to.
-            tr("There was an error trying to save the address list to %1. Please try again.", "An error message.").arg(filename));
+            /*: An error message. %1 is a name of the file (e.g., "addrbook.csv")
```
```suggestion
            /*: An error message. %1 is a stand-in argument for the name of the file 
```

hebasto replied:
[Updated](https://github.com/bitcoin-core/gui/pull/332#issuecomment-841732322).

jarolrod commented on src/qt/addressbookpage.cpp:
```diff
@@ -295,8 +297,9 @@ void AddressBookPage::on_exportButton_clicked()
-            //: %1 is a name of the file (e.g., "addrbook.csv") that the bitcoin addresses were exported to.
-            tr("There was an error trying to save the address list to %1. Please try again.", "An error message.").arg(filename));
+            /*: An error message. %1 is a name of the file (e.g., "addrbook.csv")
+                that the bitcoin addresses were exported to. */
```
```suggestion
                we attempted to export to. */
```

hebasto replied:
[Updated](https://github.com/bitcoin-core/gui/pull/332#issuecomment-841732322).

jarolrod reviewed (commented):
Concept ACK

hebasto commented:
Updated 2b28b0061823162a14c29debaaaf0312bcaa12b3 -> 8b7713365134022475e7e5d24d3ca73149bd10e1 ([pr332.01](https://github.com/hebasto/gui/commits/pr332.01) -> [pr332.02](https://github.com/hebasto/gui/commits/pr332.02), [diff](https://github.com/hebasto/gui/compare/pr332.01..pr332.02)):

- addressed @jarolrod's comments

jarolrod commented on src/qt/addressbookpage.cpp:
```diff
@@ -281,7 +281,9 @@ void AddressBookPage::on_exportButton_clicked()
     QString filename = GUIUtil::getSaveFileName(this,
         tr("Export Address List"), QString(),
-        tr("Comma separated file", "Name of CSV file format") + QLatin1String(" (*.csv)"), nullptr);
+        /*: Expanded name of the CSV file format.
```
How do we avoid having to translate this twice, same with PSBT?

hebasto replied:
We code avoid code duplication by making constants. But not sure if it worth.

jarolrod reviewed (commented):
ACK 8b7713365134022475e7e5d24d3ca73149bd10e1

From the top of the repo, I grepped with:

```
grep -rn "tr(\".*\", \".*\".*)" *
```

manually filtering out the shortlist of results, I end up with the same occurrences being addressed here:

```
src/qt/addressbookpage.cpp:284:        tr("Comma separated file", "Name of CSV file format") + QLatin1String(" (*.csv)"), nullptr);
src/qt/addressbookpage.cpp:299:            tr("There was an error trying to save the address list to %1. Please try again.", "An error message.").arg(filename));
src/qt/walletview.cpp:276:        tr("Wallet Data", "Name of wallet data file format") + QLatin1String(" (*.dat)"), nullptr);
src/qt/sendcoinsdialog.cpp:435:                tr("Partially Signed Transaction (Binary)", "Name of binary PSBT file format") + QLatin1String(" (*.psbt)"), &selectedFilter);
src/qt/transactionview.cpp:339:        tr("Comma separated file", "Name of CSV file format") + QLatin1String(" (*.csv)"), nullptr);
src/qt/psbtoperationsdialog.cpp:144:        tr("Partially Signed Transaction (Binary)", "Name of binary PSBT file format") + QLatin1String(" (*.psbt)"), &selected_filter);
src/qt/intro.cpp:388:    ui->lblPruneSuffix->setText(tr("(sufficient to restore backups %n day(s) old)", "block chain pruning", expected_backup_days));
src/qt/qrimagewidget.cpp:121:        tr("PNG Image", "Name of PNG file format") + QLatin1String(" (*.png)"), nullptr);
```
